ui-v2/tests/steps.js

```diff
@@ -0,0 +1,232 @@
+      })
+      .then('I fill in "$name" with "$value"', function(name, value) {
+        return currentPage.fillIn(name, value);
+      })
+      .then(['I fill in with yaml\n$yaml', 'I fill in with json\n$json'], function(data) {
+        return Object.keys(data).reduce(function(prev, item, i, arr) {
+          return prev.fillIn(item, data[item]);
+        }, currentPage);
+      })
+      .then(['I type with yaml\n$yaml'], function(data) {
+        const keys = Object.keys(data);
+        return keys
+          .reduce(function(prev, item, i, arr) {
+            return prev.fillIn(item, data[item]);
+          }, currentPage)
+          .then(function() {
+            return Promise.all(
+              keys.map(function(item) {
+                return triggerKeyEvent(`[name="${item}"]`, 'keyup', 83);
+              })
+            );
+          });
+      })
+      // debugging helpers
+      .then('print the current url', function(url) {
+        console.log(currentURL());
+        return Promise.resolve();
+      })
+      .then('log the "$text"', function(text) {
+        console.log(text);
+        return Promise.resolve();
+      })
+      .then('pause for $milliseconds', function(milliseconds) {
+        return new Promise(function(resolve) {
+          setTimeout(resolve, milliseconds);
+        });
+      })
+      // assertions
+      .then('a $method request is made to "$url" with the body from yaml\n$yaml', function(
+        method,
+        url,
+        data
+      ) {
+        const request = api.server.history[api.server.history.length - 2];
+        assert.equal(
+          request.method,
+          method,
+          `Expected the request method to be ${method} but was ${request.method}`
+        );
+        assert.equal(
+          request.url,
+          url,
+          `Expected the request url to be ${url} but was ${request.url}`
+        );
+        const body = JSON.parse(request.requestBody);
+        Object.keys(data).forEach(function(key, i, arr) {
+          assert.equal(
+            body[key],
+            data[key],
+            `Expected the payload to contain ${key} to equal ${body[key]} but ${key} was ${
+              data[key]
+            }`
+          );
+        });
+      })
+      .then('a $method request is made to "$url" with the body "$body"', function(
+        method,
+        url,
+        data
+      ) {
+        const request = api.server.history[api.server.history.length - 2];
+        assert.equal(
+          request.method,
+          method,
+          `Expected the request method to be ${method} but was ${request.method}`
+        );
+        assert.equal(
+          request.url,
+          url,
+          `Expected the request url to be ${url} but was ${request.url}`
+        );
+        const body = request.requestBody;
+        assert.equal(
+          body,
+          data,
+          `Expected the request body to be ${body} but was ${request.requestBody}`
+        );
```

Comment:
This assertion can fail if the two bodies are ordered differently but functionally equivalent.

Reply:
Realized `deepEqual` won't work for me here, I want to be able to match against a partial object (so I might not want to specify the entire payload in my test fixture). I think you've made a good point though, if/when it fails because of the wrong order I'll be aware and can sort it then. For testing purposes it doesn't matter that the properties in the payload are in the right or wrong order so what it does at the moment is fine for me right now.